renku/core/models/datasets.py

```diff
@@ -615,6 +617,130 @@ def __attrs_post_init__(self):
                 self.name, self.version
             )
 
+    @classmethod
+    def from_yaml(cls, path, client=None, commit=None):
+        """Return an instance from a YAML file."""
+        from marshmallow import INCLUDE
+
+        data = jsonld.read_yaml(path)
+
+        extra = dict(client=client, commit=commit)
+        data.update(extra)
+        self = DatasetSchema().load(data, unknown=INCLUDE)
```

Comment:
For the future, it'd be possible to subclass the calamus JsonldSchema so it supports `client` and `commit` as parameters and passes them on to children if they are set (saving us quite a bit of headache). What do you think? We can do a followup issue.

Reply:
That's a good idea! I've done it.